### PR TITLE
Feat: Creating API/fetching in card status successfully Issue 365

### LINKED ISSUE #365: [Feature Request]: Creating API/fetching in card status
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

1. In this i will create a API for fetching the total users and api which calculate the total posts in style share uploaded till now.
2. and showing in the style card at main dashboard page.
3. Also i will add extra card which calculate the visits and page view i will rename it.
4. Page views => when user reload the page.
5. Visits => when user will open in new tab. (For this i have to create two api)
6. Total Api's will be 4.

### Use Case

Admin Analytics

### Benefits

Tracking

### Add ScreenShots

Eg:

image

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by MeetDOD**
# Pull Request Resolves [#365 ]

### Description

1.  I created a controller and route for getting the total counts for totalUsers, totalPosts, totalComments, totalReactions, messages, favorites.
2.  And I am fetching it in the StatusCard.
3.  I had used appropriate icons for the cards.
4.  Here the messages are those which are sended from the contact us form. 

### Related Issues
Fixes #365 

### Changes Made
New features: Creating API/fetching in card status.

### Screenshots

image

image

I certify that I have carried out the relevant code of conduct and provided the requisite screenshot for validation by submitting this pull request.

Thank You for this contribution.

### CODE REVIEW

File: admin/src/components/StatusCard.tsx [hunk 1]
```diff
@@ -1,50 +1,110 @@
...
+import { tokenState } from '../store/atoms/auth';
+import { useRecoilValue } from 'recoil';
+import { MdAddReaction } from "react-icons/md";
+import { SiGooglemessages } from "react-icons/si";
+import { RiHeartsFill } from "react-icons/ri";
+import { FaComments } from "react-icons/fa6";
+import axios from 'axios';
 
 const StatusCard = () => {
+  const [stats, setStats] = useState({ totalUsers: 0, totalPosts: 0, totalComments: 0, totalReactions: 0, messages: 0, favorites: 0 });
```
Comment (akbatra567) [inline admin/src/components/StatusCard.tsx:14]: You can use a variable to save initial stats.

```suggestion
  const initialStats = { totalUsers: 0, totalPosts: 0, totalComments: 0, totalReactions: 0, messages: 0, favorites: 0 };
  const [stats, setStats] = useState(initialStats);
```

File: admin/src/components/StatusCard.tsx [hunk 2]
```diff
@@ -1,50 +1,110 @@
...
+  useEffect(() => {
+    const fetchStats = async () => {
+      try {
+        const response = await axios.get('/api/v1/admin/getCardStatus', {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        });
+        setStats(response.data);
+        console.log(response.data);
```
Comment (akbatra567) [inline admin/src/components/StatusCard.tsx:26]: remove the console log statement.

**Comment by VaibhavArora314** [comment]
rename messages to contact messages, the rest is good

**Review by VaibhavArora314 — APPROVED** [review]

**Comment by MeetDOD** [comment]
@VaibhavArora314 Done the changes 👍

File: admin/src/components/StatusCard.tsx (hunk 2 above)
Comment (MeetDOD) [reply admin/src/components/StatusCard.tsx]: Done

File: admin/src/components/StatusCard.tsx (hunk 1 above)
Comment (MeetDOD) [reply admin/src/components/StatusCard.tsx]: Completed !

**Review by Ultimateutkarsh11 — COMMENTED** [review]
@MeetDOD kindly resolve the merge conflicts

**Comment by MeetDOD** [comment]
> @MeetDOD kindly resolve the merge conflicts

Done please add level 3 label because of api + frontend 😄

**Comment by Ultimateutkarsh11** [comment]
> > @MeetDOD kindly resolve the merge conflicts
> 
> Done please add level 3 label because of api + frontend 😄

Merge conflicts are still present

**Comment by MeetDOD** [comment]
> > > @MeetDOD kindly resolve the merge conflicts
> > 
> > 
> > Done please add level 3 label because of api + frontend 😄
> 
> Merge conflicts are still present

ya its happened because you merge the 2 PR of admin,
Done @Ultimateutkarsh11 !

**Review by Ultimateutkarsh11 — APPROVED** [review]

**Comment by MeetDOD** [comment]
> > @MeetDOD kindly resolve the merge conflicts
> 
> Done please add level 3 label because of api + frontend 😄

@Ultimateutkarsh11 please consider my request if possible 👍

**Pull request merged**